packages/svelte/src/internal/client/reactivity/effects.js

```diff
@@ -217,22 +231,119 @@ export function render_effect(fn, block = current_block, managed = false, sync =
 }
 
 /**
- * @param {import('#client').Effect} signal
+ * @param {import('#client').Effect} effect
  * @returns {void}
  */
-export function destroy_effect(signal) {
-	destroy_children(signal);
-	remove_reactions(signal, 0);
-	set_signal_status(signal, DESTROYED);
-
-	signal.teardown?.();
-	signal.ondestroy?.();
-	signal.fn =
-		signal.effects =
-		signal.teardown =
-		signal.ondestroy =
-		signal.ctx =
-		signal.block =
-		signal.deps =
+export function destroy_effect(effect) {
+	destroy_children(effect);
+	remove_reactions(effect, 0);
+	set_signal_status(effect, DESTROYED);
+
+	if (effect.transitions) {
+		for (const transition of effect.transitions) {
+			transition.stop();
+		}
+	}
+
+	effect.teardown?.();
+	effect.ondestroy?.();
+
+	// @ts-expect-error
+	effect.fn =
+		effect.effects =
+		effect.teardown =
+		effect.ondestroy =
+		effect.ctx =
+		effect.block =
+		effect.deps =
 			null;
 }
+
+/**
+ * @param {import('#client').Effect} effect
+ * @param {() => void} callback
+ */
+export function pause_effect(effect, callback = noop) {
+	/** @type {import('#client').TransitionManager[]} */
+	const transitions = [];
+
+	pause_children(effect, transitions, true);
+
+	let remaining = transitions.length;
+
+	if (remaining > 0) {
+		const check = () => {
+			if (!--remaining) {
+				destroy_effect(effect);
+				callback();
+			}
+		};
+
+		for (const transition of transitions) {
+			transition.out(check);
+		}
+	} else {
+		destroy_effect(effect);
+		callback();
+	}
+}
+
+/**
+ * @param {import('#client').Effect} effect
```

Comment:
what does pausing children mean? why is it important? -> docs (could reference pause_effect docs)